tests/func/test_s3.py

```diff
@@ -111,3 +111,17 @@ def test_copy_multipart_preserve_etag():
     s3.create_bucket(Bucket=from_info.bucket)
     _upload_multipart(s3, from_info.bucket, from_info.path)
     S3Tree._copy(s3, from_info, to_info, {})
+
+
+@mock_s3
+def test_s3_isdir(tmp_dir, dvc, scm):
+    path_info = S3Tree.PATH_CLS("s3://bucket/data/foo")
+    tree = S3Tree(dvc, {"url": str(path_info.parent)})
+
+    s3 = tree.s3.meta.client
+    s3.create_bucket(Bucket=path_info.bucket)
+    s3.put_object(Bucket=path_info.bucket, Key=path_info.path, Body="data")
+
+    tree.makedirs(path_info.parent)
+    assert not tree.isdir(path_info)
+    assert tree.isdir(path_info.parent)
```

Comment:
The tests above are a bit obsolete, we have a more handy way to do this now:
```suggestion
def test_s3_isdir(tmp_dir, dvc, s3):
    s3.gen({"data": {"foo": "foo"}})
    tree = S3Tree(dvc, s3.config)
    
    assert not tree.isdir(s3 / "data" / "foo")
    assert tree.isdir(s3 / "data")
```
also please avoid using `scm` fixture unless you need git, as it slows down the test.

Reply:
Thanks for the tips!